provision/ec2.go

```diff
@@ -165,39 +166,70 @@ func (p *EC2Provisioner) CreateInstance(opts EC2CreateInstanceOptions) (*cfg.Rem
 		return nil, err
 	}
 
+	// Check response validity
+	if runResp.Instances == nil || len(runResp.Instances) == 0 {
+		return nil, errors.New("Unable to start instances: " + runResp.String())
+	}
+
 	// Loop until intance is running
-	fmt.Fprint(p.out, "Checking status of requested instance...")
-	attempts := 0
-	var instanceStatus *ec2.DescribeInstancesOutput
+	fmt.Fprintln(p.out, "Checking status of requested instance...")
+	var (
+		attempts = 0
+		instance ec2.Instance
+	)
 	for {
+		// Wait briefly between checks
+		time.Sleep(3 * time.Second)
 		attempts++
+
 		// Request instance status
 		result, err := p.client.DescribeInstances(&ec2.DescribeInstancesInput{
 			InstanceIds: []*string{runResp.Instances[0].InstanceId},
 		})
 		if err != nil {
 			return nil, err
 		}
-		if len(result.Reservations) == 0 || len(result.Reservations[0].Instances) == 0 {
+
+		// Check if reservations are present
+		if result.Reservations == nil || len(result.Reservations) == 0 ||
+			len(result.Reservations[0].Instances) == 0 {
 			// A reservation corresponds to a command to start instances
-			time.Sleep(3 * time.Second)
+			// If nothing is here... we gotta keep waiting
+			fmt.Fprintln(p.out, "No reservations found yet.")
+			continue
+		}
+
+		// Get status
+		s := result.Reservations[0].Instances[0].State
+		if s == nil {
+			fmt.Println(p.out, "Status unknown.")
 			continue
-		} else if *result.Reservations[0].Instances[0].State.Code == 16 {
-			// Code 16 means we can continue!
-			fmt.Fprint(p.out, "Instance is running!")
-			instanceStatus = result
+		}
+
+		// Code 16 means instance has started, and we can continue!
+		if s.Code != nil && *s.Code == 16 {
```

Comment:
We could make a global variable called `EC2_INSTANCE_STARTED_CODE` or something. Then we'd have self-documenting code! 😄

Reply:
Why global? The code is part of the API

Reply:
I feel like the AWS library should be the one that declares this. Oh well

Reply:
<img width="416" alt="image" src="https://user-images.githubusercontent.com/23356519/43679873-58de9682-97e2-11e8-9d2b-413f900694a6.png">